src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/GoCardlessMandateHandlerTest.java

```diff
@@ -49,185 +48,173 @@
 
-        verify(mockedTransactionService).paymentFailedWithoutEmailFor(transaction);
-        verify(mockedMandateService).mandateCancelledFor(transaction, payer);
-        verify(goCardlessEvent).setPaymentRequestEventId(paymentRequestEvent.getId());
-        verify(mockedGoCardlessService).storeEvent(geCaptor.capture());
+        verify(mockedTransactionService).paymentFailedWithoutEmailFor(transactionFixture.toEntity());
+        verify(mockedMandateService).mandateCancelledFor(mandateFixture.toEntity());
+        verify(goCardlessEvent).setEventId(event.getId());
+        verify(mockedGoCardlessService).updateInternalEventId(geCaptor.capture());
         GoCardlessEvent storedGoCardlessEvent = geCaptor.getValue();
-        Assert.assertThat(storedGoCardlessEvent.getPaymentRequestEventId(), is(paymentRequestEvent.getId()));
+        Assert.assertThat(storedGoCardlessEvent.getEventId(), is(event.getId()));
+        Assert.assertThat(storedGoCardlessEvent.getGoCardlessEventId(), is(goCardlessEvent.getGoCardlessEventId()));
     }
 
     @Test
     public void handle_onACancelledMandateGoCardlessEvent_shouldRegisterAPayEventAsMandateCancelled_withoutFailingPaymentIfSubmittedToGC() {
         GoCardlessEvent goCardlessEvent = spy(GoCardlessEventFixture.aGoCardlessEventFixture().withAction("cancelled").toEntity());
 
         when(mockedGoCardlessService.findMandateForEvent(goCardlessEvent)).thenReturn(goCardlessMandateFixture.toEntity());
-        when(mockedPayerService.getPayerFor(transaction)).thenReturn(payer);
-        when(mockedTransactionService.findPaymentSubmittedEventFor(transaction)).thenReturn(Optional.of(paymentRequestEvent));
-        when(mockedMandateService.mandateCancelledFor(transaction, payer)).thenReturn(paymentRequestEvent);
+        when(mockedTransactionService.findPaymentSubmittedEventFor(transactionFixture.toEntity())).thenReturn(Optional.of(
+                event));
+        when(mockedMandateService.mandateCancelledFor(mandateFixture.toEntity())).thenReturn(event);
 
         goCardlessMandateHandler.handle(goCardlessEvent);
 
-        verify(mockedTransactionService, never()).paymentFailedWithoutEmailFor(transaction);
-        verify(mockedMandateService).mandateCancelledFor(transaction, payer);
-        verify(goCardlessEvent).setPaymentRequestEventId(paymentRequestEvent.getId());
-        verify(mockedGoCardlessService).storeEvent(geCaptor.capture());
+        verify(mockedTransactionService, never()).paymentFailedWithoutEmailFor(transactionFixture.toEntity());
+        verify(mockedMandateService).mandateCancelledFor(mandateFixture.toEntity());
+        verify(mockedGoCardlessService).updateInternalEventId(geCaptor.capture());
         GoCardlessEvent storedGoCardlessEvent = geCaptor.getValue();
-        Assert.assertThat(storedGoCardlessEvent.getPaymentRequestEventId(), is(paymentRequestEvent.getId()));
-    }
-
-    @Test
-    public void shouldStoreEventsWithAnUnhandledAction() {
```

Comment:
Not sure why this was removed

Reply:
because storing events at this level was unnecessary as it has been already stored in WebhookGoCardlessService (line 33). Too lazy to copy the link.